default-with-analytics/WSO2AM210-All-In-One-Deployment-VPC.template.yaml

```diff
@@ -235,126 +262,94 @@ Resources:
       KeyName: !Ref KeyPairName
       SecurityGroups:
         - !Ref WSO2APIMSecurityGroup
-      IamInstanceProfile: !Ref WSO2APIMAWSCLIInstanceProfile
-      UserData: !Base64 
+      UserData: !Base64
         'Fn::Join':
           - |+
 
           - - '#!/bin/bash'
             - set -e
             - 'export PATH=~/.local/bin:$PATH'
-            - !Join 
-              - ''
-              - - sed -i "s/CF_ENI_ID/
-                - !Ref WSO2APIMNode1NetworkInterface
-                - /g" /usr/local/bin/allocate_ips.sh
-            - !Join 
-              - ''
-              - - sed -i "s/CF_ALLOC_ID/
-                - !GetAtt 
-                  - WSO2APIMNode1EIP
-                  - AllocationId
-                - /g" /usr/local/bin/allocate_ips.sh
-            - >-
-              sed -i "s/CF_DEFAULT_GATEWAY/10.0.254.1/g"
-              /usr/local/bin/allocate_ips.sh
-            - !Join 
-              - ''
-              - - sed -i "s/CF_ENI_PRIVATE_IP/
-                - !GetAtt 
-                  - WSO2APIMNode1NetworkInterface
-                  - PrimaryPrivateIpAddress
-                - /g" /usr/local/bin/allocate_ips.sh
             - mkdir -p ~/.aws
             - cat > ~/.aws/config << EOF
             - '[default]'
-            - !Join 
+            - !Join
               - ''
               - - 'region = '
                 - !Ref 'AWS::Region'
             - output = text
             - EOF
-            - bash /usr/local/bin/allocate_ips.sh
             - mkdir -p /mnt/efs
-            - !Join 
-              - ''
-              - - echo "
-                - !Ref WSO2APIMEFSFileSystem
-                - .efs.
-                - !Ref 'AWS::Region'
-                - .amazonaws.com
-                - >-
-                  :/ /mnt/efs nfs4
-                  nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2
-                  0 0" >> /etc/fstab
-            - !Join 
+            - apt-get update
+            - apt-get install -y nfs-common
+            - !Sub "mount -t nfs4 -o nfsvers=4.1 ${WSO2APIMEFSFileSystem}.efs.${AWS::Region}.amazonaws.com:/ /mnt/efs"
+            - !Join
               - ''
               - - sed -i "s/CF_ELB_DNS_NAME/
-                - !GetAtt 
+                - !GetAtt
                   - WSO2APIMLoadBalancer
                   - DNSName
                 - >-
                   /g"
                   /etc/puppet/hieradata/dev/wso2/wso2am_runtime/pattern-2/default.yaml
-            - !Join 
+            - !Join
               - ''
               - - sed -i "s/CF_APIM_IP/
-                - !GetAtt 
-                  - WSO2APIMNode1NetworkInterface
-                  - PrimaryPrivateIpAddress
+                - 'localhost'
                 - >-
                   /g"
                   /etc/puppet/hieradata/dev/wso2/wso2am_runtime/pattern-2/default.yaml
-            - !Join 
+            - apt-get install -y python3-pip
```

Comment:
Better to collect all `apt-get installs` together.

Reply:
Agreed, will make the necessary changes as suggested in a future version as CloudFormation stack creation consumes a considerable amount of time.